src/shared/components/button/button.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+type ButtonProps = {
+  children: React.ReactNode;
+  onClick?: () => void;
+  type?: 'button' | 'submit' | 'reset';
+  disabled?: boolean;
+  variant?: 'primary' | 'secondary' | 'danger';
+};
+
+const Button: React.FC<ButtonProps> = ({
+  children,
+  onClick,
+  type = 'button',
+  disabled = false,
+  variant = 'primary',
+}) => {
+  const baseStyles =
+    'py-2 px-4 font-semibold rounded-lg shadow-md focus:outline-none';
+  let variantStyles = '';
+
+  switch (variant) {
+    case 'primary':
+      variantStyles = 'bg-blue-500 text-white hover:bg-blue-700';
```

Comment:
a lot time we repeate "text-white hover:bg-blue-700". Can we introduce something our custom clsx util?

Reply:
fixed